src/modules/order/order.subscriber.ts

```diff
@@ -0,0 +1,199 @@
+            store_id: store_id,
+            cart_id: null,
+            cart: null,
+            id: null,
+            shipping_methods: []
+          }) as Order;
+          const orderResult = await orderRepo.save(childOrder);
+
+          //create shipping methods
+          for (const shippingMethod of order.shipping_methods) {
+            const newShippingMethod = shippingMethodRepo.create({
+              ...shippingMethod,
+              id: null,
+              cart_id: null,
+              cart: null,
+              order_id: orderResult.id
+            });
+
+            await shippingMethodRepo.save(newShippingMethod);
+          }
+
+          //create line items
+          const items: LineItem[] = groupedItems[store_id];
+          for (const item of items) {
+            const newItem = lineItemRepo.create({
+              ...item,
+              id: null,
+              order_id: orderResult.id,
+              cart_id: null
+            })
+            await lineItemRepo.save(newItem);
+          }
+        }
+    }
+
+    public async checkStatus({ id }: {id: string}): Promise<void> {
+      //retrieve order
+      const order: Order = await this.orderService.retrieve(id);
+
+      if (order.parent_id) {
+        //retrieve parent
+        const orderRepo = this.manager.getCustomRepository(this.orderRepository);
+        const parentOrder = await this.orderService.retrieve(order.parent_id, {
+            relations: ['children']
+        });
+
+        const newStatus = this.getStatusFromChildren(parentOrder);
+        if (newStatus !== parentOrder.status) {
+            switch (newStatus) {
+              case OrderStatus.CANCELED:
+                this.orderService.cancel(parentOrder.id);
+                break;
+              case OrderStatus.ARCHIVED:
+                this.orderService.archive(parentOrder.id);
+                break;
+              case OrderStatus.COMPLETED:
+                this.orderService.completeOrder(parentOrder.id);
+                break;
+              default:
+                parentOrder.status = newStatus;
+                parentOrder.fulfillment_status = newStatus;
+                parentOrder.payment_status = newStatus;
+                await orderRepo.save(parentOrder);
+            }
+        }
+      }
+    }
+
+  public getStatusFromChildren (order: Order): string {
+      if (!order.children) {
+          return order.status;
+      }
+
+      //collect all statuses
+      let statuses = order.children.map((child) => child.status);
+
+      //remove duplicate statuses
+      statuses = [...new Set(statuses)];
+
+      if (statuses.length === 1) {
+          return statuses[0];
+      }
+
+      //remove archived and canceled orders
+      statuses = statuses.filter((status) => status !== OrderStatus.CANCELED && status !== OrderStatus.ARCHIVED);
+
+      if (!statuses.length) {
+          //all child orders are archived or canceled
+          return OrderStatus.CANCELED;
```

Comment:
here you are not sure if it is `CANCELED` or `ARCHIVED` since they are both filtered above. maybe we should not filter them? wdyt?

Reply:
something like [].every(status === canceled) then the parent order is canceled and the same approach for archived?

Reply:
Thing is it's not going to be all canceled or all archived since we've already confirmed at this point that the `statuses` array contains unique statuses and it has more than 1 element. I've mentioned in the tutorial that this is one approach and that can be changed based on the desired logic (for example they might want to label the order complete anyway).

Reply:
ok in that case we can let them choose 👍